### ISSUE TITLE
Update min LibCST version when it supports Python 3.9 relaxed decorator syntax

### ISSUE THREAD

**Issue by Zac-HD** [Zac-HD/shed#18]
```python
with (a() as b, c() as d):
    pass

@(lambda f: f)
def g():
    pass
```

is valid Python 3.9 syntax which triggers https://github.com/psf/black/issues/1948 and, in `--refactor` mode, https://github.com/Instagram/LibCST/issues/486 and https://github.com/Instagram/LibCST/issues/490.  Once these upstream bugs are fixed, we should:

- [x] update the minimum version of Black
- [x] update the minimum version of LibCST

**Comment by Zac-HD** [comment]
Per https://github.com/Instagram/LibCST/issues/285#issuecomment-1011427731, an upcoming release will enable their new Rust-based parser for 3.10 & 3.11 grammar by default :tada: 

(maybe this will also make it faster?)

**Comment by isidentical** [comment]
`black` also now supports 3.10+ syntax with patma / parenthesized with statements.

**Comment by Zac-HD** [comment]
Black now works, so we're just waiting for enabled-by-default `native` parser in libCST :tada:

**Comment by Zac-HD** [comment]
This all works now; we just need to add some tests and bump the minimum required version.

**Comment by jpaniagua26** [comment]
I tried adding a test to check for python 3.9 syntax but apparently libcst has not yet added syntax checking for this python version.  python39syntax

**Comment by jpaniagua26** [comment]
I tried adding a test to check for python 3.9 syntax but apparently libcst has not yet added syntax checking for this python version.  [https://github.com/jpaniagua26/shed/tree/python39syntax](url)

**Comment by jpaniagua26** [comment]
I tried adding adding a test to check for python 3.9 syntax but it failed. Apparently libcst has not yet added syntax checking for this python version. https://github.com/jpaniagua26/shed/tree/python39syntax

**Comment by Zac-HD** [comment]
Closed by https://github.com/Zac-HD/shed/commit/69063295a69afb1e0b9c78798cd428899fc1e590, released as `shed == 23.05.2` 🎉
